drivers/gpio/gpio_kscan_ite_it8xxx2.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2022 ITE Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_gpiokscan
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_kscan_it8xxx2.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/dt-bindings/gpio/ite-it8xxx2-gpio.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/types.h>
+
+/* Port counts: KSI[7:0], KSO[15:8] and KSO[7:0] */
+#define GPIO_KSCAN_PORT_COUNT 3
+
+/* KSI[7:0]/KSO[15:8]/KSO[7:0] port module instances */
+#define IT8XXX2_GPIO_KSCAN_DEV(inst) DEVICE_DT_INST_GET(inst),
+static const struct device *const gpio_kscan_devs[] = {
+	DT_INST_FOREACH_STATUS_OKAY(IT8XXX2_GPIO_KSCAN_DEV)
+};
+
+struct gpio_kscan_cfg {
+	/* The gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio output enable register (bit mapping to pin) */
+	volatile uint8_t *reg_ksi_kso_goen;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio control register (bit mapping to pin) */
+	volatile uint8_t *reg_ksi_kso_gctrl;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio data register (bit mapping to pin) */
+	volatile uint8_t *reg_ksi_kso_gdat;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio data mirror register (bit mapping to pin) */
+	volatile uint8_t *reg_ksi_kso_gdmr;
+	/* KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio open drain register (bit mapping to pin) */
+	volatile uint8_t *reg_ksi_kso_gpod;
+	/* Index for KSI[7:0], KSO[15:8] and KSO[7:0] port */
+	uint8_t index;
+};
+
+struct gpio_kscan_data {
+	/* The gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+};
+
+/* Platform get specific device of KSI[7:0]/KSO[15:8]/KSO[7:0] port */
+const struct device *it8xxx2_get_gpio_kscan_dev(int port)
+{
+	if (port >= ARRAY_SIZE(gpio_kscan_devs)) {
+		return NULL;
+	}
+
+	return gpio_kscan_devs[port];
```

Comment:
`DT_INST_FOREACH_STATUS_OKAY` does not guarantee order, to it's possible that `gpio_kscan_dev[0].index != 0`.

This function should not be needed.  Any code that needs to find the device handle should use `DEVICE_DT_GET(DT_NODELABEL(gpioksi))`.

Reply:
Ack, thanks!
I removed.